contracts/NodeVersionBeacon.cdc

```diff
@@ -280,6 +285,13 @@ access(all) contract NodeVersionBeacon {
 
             emit NodeVersionBoundaryFreezePeriodChanged(freezePeriod: newFreezePeriod)
         }
+
+        /// Emits the given protocol state version upgrade event. If the version and active are
+        /// valid, this will cause the Protocol State to upgrade its model version when the
+        /// event is incorporated.
+        access(all) fun emitProtocolStateVersionUpgrade(newProtocolVersion: UInt64, activeView: UInt64) {
```

Comment:
Is there any validation we can do on these inputs? Like the view needs to be in the future, version needs to be greater than the current version, etc?

Reply:
We can check that view needs to be in the future (but the actual acceptance criteria is more involved). We can't check the version is greater than current, because the current version only exists in the protocol state. 

The complete validation is done in the Protocol State Machine, where the service event is processed. Even if we partially replicate the validation here, we still need to wait for the service event to be processed at the Protocol level before we know it was successful. 

Since emitting these events is not state-changing from the smart contract perspective (you can emit as many as you want, it doesn't change any on-chain state), I'm inclined to keep all the validation only in the Protocol State machine. I'll add some documentation to explain this.